conftest.py

```diff
@@ -111,156 +276,87 @@ def temp_dir():
-    # Convert python-specific data objects (such as tuples) to a more
-    # io-friendly format (in order to not have python-specific anchors
-    # in the answer yaml file)
-    test_params = utils._streamline_for_io(test_params)
-    return test_params
+    Fixture for returning the ds_obj. Needed because indirect=True is
+    used for loading the datasets.
+    """
+    return request.param
 
 
-@pytest.fixture(scope="function")
-def hashing(request):
-    r"""
-    Handles initialization, generation, and saving of answer test
-    result hashes.
+@pytest.fixture(scope="class")
+def a(request):
+    """
+    Fixture for returning the axis. Needed because indirect=True is
+    used for loading the datasets.
+    """
+    return request.param
 
-    Answer tests that require generated data to be saved to disk
-    have that data hashed. This fixture creates an empty dictionary
-    as an attribute of the answer class of which the current answer
-    test is a method.
-
-    Once the test has been run and the raw data has been saved to this
-    hashes dictionary, this fixture hashes the raw data and prepares
-    an entry to the answer file containing the test name as well as the
-    test parameter names and values to accompany the hash(es).
-
-    These entries are then either compared to an existing entry or
-    saved in the new answer file.
-
-    Parameters:
-    -----------
-        request : pytest.FixtureRequest
-            Provides access to the requesting test context. For
-            example, if an answer class uses this fixture, such as
-            TestEnzo, then request provides access to all of the
-            methods and attributes of the TestEnzo class, since
-            that class is the user of this fixture (the calling
-            context).
-
-    Example:
-    --------
-        # If every method of an answer class saves data then the
-        # fixture can be applied to each method like so:
-        >>> @pytest.mark.usefixtures('hashing')
-        >>> class TestNewFrontend:
-        >>>     def test1(self):
-        >>>         ...
-        >>>     def test2(self):
-        >>>         ...
-        # If only certain methods save data, then it must be applied
-        # directly to those methods, like so:
-        >>> class TestNewFrontend:
-        >>>     @pytest.mark.usefixtures('hashing')
-        >>>     def test1(self):
-        >>>         ...
-        >>>     def test2(self):
-        >>>         ...
+
+@pytest.fixture(scope="class")
+def w(request):
+    """
+    Fixture for returning the weight_field. Needed because
+    indirect=True is used for loading the datasets.
+    """
+    return request.param
+
+
+@pytest.fixture(scope="class")
+def ds_repr(request):
+    """
+    Fixture for returning the string representation of a dataset.
+    Needed because indirect=True is used for loading the datasets.
+    """
+    return request.param
+
+
+@pytest.fixture(scope="class")
+def N(request):
```

Comment:
```suggestion
def Npart(request):
```
or
```suggestion
def number_of_particles(request):
```
Also if there are different particle types, which one is returned?

Reply:
This is currently only used by arepo when checking the total number of particles in the sph test, so it doesn't deal with the different particle types at all because it currently doesn't need to.